server/embed/etcd.go

```diff
@@ -611,6 +614,9 @@ func configureClientListeners(cfg *Config) (sctxs map[string]*serveCtx, err erro
 	if err = cfg.ClientSelfCert(); err != nil {
 		cfg.logger.Fatal("failed to get client self-signed certs", zap.Error(err))
 	}
+	if err = updateMinMaxVersions(&cfg.ClientTLSInfo, cfg.TlsMinVersion, cfg.TlsMaxVersion); err != nil {
+		return nil, err
+	}
```

Comment:
This side effect approach of generating TLSInfo is very error prone. Are we sure we called `updateMinMaxVersions` in all places that TLSInfo is changed?


Would really like to have this code cleaned up.

Reply:
I copied the approach from how the ciphers seemed to be generated to `TLSInfo` by similar `updateCipherSuites()` function (called [here](https://github.com/etcd-io/etcd/blob/main/server/embed/config.go#L932) and [here]( https://github.com/etcd-io/etcd/blob/3b612ce345753f97156833e20e93bf78b56d5267/server/embed/config.go#L951)). I'm little doubtful if I'm able to do bigger refactoring for `TLSInfo` generation. Maybe there is an approach to clean up but with minimal impact to existing code?

Reply:
Sounds good.